src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/IslHelper.groovy

```diff
@@ -28,25 +37,37 @@ class IslHelper {
     IslUtils islUtils
     @Autowired
     PortAntiflapHelper antiflapHelper
+    @Autowired
+    CleanupManager cleanupManager
+    @Autowired
+    Database database
+    @Autowired
+    TopologyDefinition topology
+    @Autowired @Qualifier("islandNb")
+    NorthboundService northbound
+    @Autowired @Qualifier("islandNbV2")
+    NorthboundServiceV2 northboundV2
 
 
-    def breakIsl(Isl islToBreak) {
+    def breakIsl(Isl islToBreak, CleanupAfter cleanupAfter = TEST) {
+        cleanupManager.addAction(RESTORE_ISL,{restoreIsl(islToBreak)}, cleanupAfter)
+        cleanupManager.addAction(RESET_ISLS_COST,{database.resetCosts(topology.isls)}, cleanupAfter)
         if (getIslStatus(islToBreak).equals(DISCOVERED)) {
             antiflapHelper.portDown(islToBreak.getSrcSwitch().getDpId(), islToBreak.getSrcPort())
         }
         islUtils.waitForIslStatus([islToBreak], FAILED)
     }
 
-    def breakIsls(Set<Isl> islsToBreak) {
+    def breakIsls(Set<Isl> islsToBreak, CleanupAfter cleanupAfter = TEST) {
         withPool {
             islsToBreak.eachParallel{
-                breakIsl(it)
+                breakIsl(it, cleanupAfter)
```

Comment:
It seems like we call `database.resetCosts(topology.isls)` for all ISls for each broken ISL during restoring in the scope of cleanupManager.

Reply:
No, in CleanupManager we only reset all isls cost once:
```
 if (actions[RESET_ISLS_COST]) {
            exceptions += runActionsSynchronously([actions[RESET_ISLS_COST].first()])
 }
 ```